earth2studio/models/dx/solarradiation_afno.py

```diff
@@ -0,0 +1,363 @@
+                )
+            )
+        ).permute(1, 0, 2, 3)
+        z = (z - z.mean()) / z.std()
+
+        lsm = torch.Tensor(
+            np.load(
+                str(
+                    Path(
+                        package.resolve(
+                            f"ssrd_{freq}_afno/solarradiation_afno/land_sea_mask.npy"
+                        )
+                    )
+                )
+            )
+        ).permute(1, 0, 2, 3)
+
+        sincos_latlon = torch.Tensor(
+            np.load(
+                str(
+                    Path(
+                        package.resolve(
+                            f"ssrd_{freq}_afno/solarradiation_afno/sincos_latlon.npy"
+                        )
+                    )
+                )
+            )
+        ).permute(1, 0, 2, 3)
+        return cls(
+            model, freq, era5_mean, era5_std, ssrd_mean, ssrd_std, z, lsm, sincos_latlon
+        )
+
+    def get_sza_lonlat(
+        self, lon: np.ndarray, lat: np.ndarray
+    ) -> tuple[np.ndarray, np.ndarray]:
+        """Get longitude and latitude arrays for solar zenith angle calculation.
+
+        Parameters
+        ----------
+        lon : np.ndarray
+            Longitude array
+        lat : np.ndarray
+            Latitude array
+
+        Returns
+        -------
+        tuple[np.ndarray, np.ndarray]
+            Tuple of (longitude, latitude) arrays
+        """
+        grid = np.meshgrid(lon, lat)
+        return (grid[0].reshape(-1), grid[1].reshape(-1))
+
+    def compute_sza(self, output_coords: CoordSystem) -> torch.Tensor:
+        """Compute solar zenith angle for given coordinates.
+
+        Parameters
+        ----------
+        output_coords : CoordSystem
+            Output coordinate system
+
+        Returns
+        -------
+        torch.Tensor
+            Solar zenith angle tensor
+        """
+        lon, lat = self.get_sza_lonlat(output_coords["lon"], output_coords["lat"])
+        t = output_coords["time"] + output_coords["lead_time"]
+        t = datetime.fromtimestamp(
+            t.astype("datetime64[s]").astype("int")[0], tz=timezone.utc
+        )
+        return torch.Tensor(cos_zenith_angle(t, lon, lat))
+
+    @torch.inference_mode()
+    @batch_func()
+    def __call__(
+        self,
+        x: torch.Tensor,
+        coords: CoordSystem,
+    ) -> tuple[torch.Tensor, CoordSystem]:
+        """Forward pass of diagnostic"""
+        # Reshape to remove batch, time, and lead_time dimensions
+        batch_size = x.shape[0]
+        time_size = x.shape[1]
+        lead_time_size = x.shape[2]
+        x = x.reshape(
+            -1, *x.shape[3:]
+        )  # Combine batch, time, lead_time into one dimension
+
+        # Normalize input
+        x = (x - self.era5_mean) / self.era5_std
+        output_coords = self.output_coords(coords)
+
+        # compute solar zenith angle and concatenate
+        sza = self.compute_sza(output_coords).reshape((1, 1, *x.shape[2:])).to(x.device)
```

Comment:
With `reshape((1, 1, *x.shape[2:]))`

Doesnt this assume a time dim of size 1, should then be `reshape((1, *x.shape[1:]))` with the edit suggested above?

Reply:
Oh x is already collapsed here, but im still not following the dims of what the second 1 is, is that for variable? If so then shouldnt the first dim be time*lead_time?

Perhaps its easier to reshape to `.reshape((1, time_size, lead_time_size, *x.shape[3:]))`
Then repeat dim0 for batch_size, then reshape like x above.

Reply:
So I rewrote this to run in a loop similar to the precipitation_afno_v2. The issue you raised here would have caused issues when running on multiple time or lead times. Its fixed now though.